source/neuropod/internal/neuropod_tensor.hh

```diff
@@ -502,6 +502,44 @@ public:
         return out;
     }
 
+    // Get the data's element by index.
+    // This is method specific for String Tensor. This is added for optimization
+    // when caller needs to get one element of data. Same as if it called get_data_as_vector
+    // and then read element at index. This allows to avoid overhead of getting whole vector
+    // that is critical for tensor of a large size.
+    std::string get_data_at(size_t index) const
+    {
+        this->assure_device_cpu();
+
+        const auto numel = get_num_elements();
+        if (index >= numel)
+        {
+            NEUROPOD_ERROR_HH("The index is out of bounds in the tensor.");
+        }
+
+        // Get element by index.
+        return get(index);
+    }
+
+    // Set the data's element by index.
+    // This is method specific for String Tensor. This is added for optimization
+    // when caller needs to set one element of data at given index instead of using
+    // copy_from that requires vector of all elements. This also can be used if need to
+    // set all elements one by one without creating intermidiate vector.
+    void set_data_at(size_t index, const std::string &value)
+    {
+        this->assure_device_cpu();
+
+        const auto numel = get_num_elements();
+        if (index >= numel)
+        {
+            NEUROPOD_ERROR_HH("The index is out of bounds in the tensor.");
+        }
+
+        // Set element by index.
+        set(index, value);
+    }
+
```

Comment:
Is it some pattern flat instead of get/set? I am not familiar with it but anyway, you can publish it and I will be looking on it through review. 

I am ok to wait for this and see how it solves problem for StringTensor, I need to rebase anyway.